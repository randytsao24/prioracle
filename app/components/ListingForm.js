// app/components/ListingForm.js
// View for a user to enter product information and obtain 
// a suggested price.

import React, { Component } from 'react';
import { Alert, StyleSheet, Text, KeyboardAvoidingView, Picker, ScrollView } from 'react-native';
import { StackNavigator, NavigationActions } from 'react-navigation';
import { Card, Button, FormLabel, FormInput, FormValidationMessage, Header } from 'react-native-elements';
import { connect } from 'react-redux';

import { addListing } from '../store';

class ListingForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      productName: '',
      productDescription: '',
      productCategory: '',
      selectedCondition: 'New',
      selectedShipping: 'Buyer'
    };

    this.showErrorAlert = this.showErrorAlert.bind(this);
    this.getProductAnalysis = this.getProductAnalysis.bind(this);
  }

  showErrorAlert(inputField, userInput) {
    if (inputField === 'product-name-blank')
      Alert.alert('Product name cannot be empty!');
    else if (inputField === 'product-desc-blank')
      Alert.alert('Product description cannot be blank!');
    else if (inputField === 'product-category-blank')
      Alert.alert('Product category cannot be blank!');
    else if (inputField === 'product-category-mismatch')
      Alert.alert('Product category must match the following format:\n\nCategoryOne/CategoryTwo/CategoryThree\n' + userInput);
  }

  getProductAnalysis() {
    // Validate that user has filled out the product name field
    if (this.state.productName === '') {
      this.showErrorAlert('product-name-blank'); return;
    }
    
    // Create new listing object with our form data
    const listingObj = {
      name: this.state.productName,
      description: this.state.productDescription,
      category: this.state.productCategory,
      condition: this.state.selectedCondition,
      sellerShips: this.state.selectedShipping === 'Seller' ? new Boolean(true) : new Boolean(false),
      status: 'inactive'
    };

    const productListing = {
      listing: listingObj,
      user: this.props.user
    };

    // Submit post request with our filled-in form data
    this.props.addListingFromForm(productListing, this.props.navigation);
  }

  render() {
    return (
      <ScrollView contentContainerStyle={styles.scrollContainer}>
        <KeyboardAvoidingView 
          style={styles.container}
          behavior='padding' >

          <Card title='Add a New Listing' >
            {/* PRODUCT NAME FIELD */}
            <FormLabel labelStyle={styles.formLabel}>Product Name</FormLabel>
            <FormInput
              inputStyle={styles.inputText}
              textAlign={'center'}
              onChangeText={text => this.setState({productName: text})} />

            {/* PRODUCT DESCRIPTION FIELD */}
            <FormLabel labelStyle={styles.formLabel}>Product Description</FormLabel>
            <FormInput
              containerStyle={styles.inputContainer}
              inputStyle={styles.inputText}
              textAlign={'center'}
              multiline={true}
              onChangeText={text => this.setState({productDescription: text})} />
            
            {/* PRODUCT CATEGORY FIELD */}
            <FormLabel labelStyle={styles.formLabel}>Product Category (/-separated)</FormLabel>
            <FormInput
              inputStyle={styles.inputText}
              textAlign={'center'}
              onChangeText={text => this.setState({productCategory: text})} />

            {/* PRODUCT CONDITION FIELD */}
            <FormLabel labelStyle={styles.formLabel}>Product Condition</FormLabel>
            <Picker
              selectedValue={this.state.selectedCondition}
              onValueChange={(itemValue) => this.setState({selectedCondition: itemValue})}>
              <Picker.Item label="New" value="New" />
              <Picker.Item label="Used" value="Used" />
              <Picker.Item label="Like New" value="Like New" />
            </Picker>

            {/* USER SHIPPING FIELD */}
            <FormLabel labelStyle={styles.formLabel}>Who pays for shipping?</FormLabel>
            <Picker
              selectedValue={this.state.selectedShipping}
              onValueChange={(itemValue) => this.setState({selectedShipping: itemValue})}>
              <Picker.Item label="Buyer" value="Buyer" />
              <Picker.Item label="Seller" value="Seller" />
            </Picker>
          </Card>
          
          <Button
            buttonStyle={styles.submitButton}
            title='Crunch the numbers!'
            onPress={() => this.getProductAnalysis()} />
          <Text>{"\n"}</Text>
        </KeyboardAvoidingView>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
    container: { 
      flex: 1,
      justifyContent: 'flex-start',
      backgroundColor: '#e1e8e6'
    },
    scrollContainer: {
      backgroundColor: '#e1e8e6'
    },
    headerOuterContainer: {
      height: 50
    },
    formLabel: {
      color: 'red'
    },
    inputContainer: {
      flex: 1,
      justifyContent: 'center'
    },
    inputText: {
      flex: 1,
      color: 'black'
    },
    submitButton: {
      backgroundColor: '#d14f4f'
    }
});

const mapStateToProps = (state) => {
  return {
    user: state.user
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    addListingFromForm(listing, navigation) {
      dispatch(addListing(listing, navigation));
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(ListingForm);