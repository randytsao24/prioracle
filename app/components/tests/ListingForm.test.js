// app/components/tests/ListingForm.test.js
// Test basic functionality for our ListingForm component

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { configure, shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from '../../../App';
import ListingForm from '../ListingForm';

configure({ adapter: new Adapter() });

const initialState = {};

const mockStore = configureStore();

test('ListingForm renders correctly', () => {
  const wrapper = shallow(
    <ListingForm />,
    { context: { store: mockStore(initialState) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
});
