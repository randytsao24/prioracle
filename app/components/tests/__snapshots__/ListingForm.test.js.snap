// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingForm renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListingForm
    addListingFromForm={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <KeyboardAvoidingView
        behavior="padding"
        keyboardVerticalOffset={0}
        style={
          Object {
            "backgroundColor": "#e1e8e6",
            "flex": 1,
            "justifyContent": "flex-start",
          }
        }
      >
        <Card
          title="Add a New Listing"
        >
          <FormLabel
            labelStyle={
              Object {
                "color": "red",
              }
            }
          >
            Product Name
          </FormLabel>
          <FormInput
            inputStyle={
              Object {
                "color": "black",
              }
            }
            normalizeFontSize={true}
            onChangeText={[Function]}
            textAlign="center"
          />
          <FormLabel
            labelStyle={
              Object {
                "color": "red",
              }
            }
          >
            Product Description
          </FormLabel>
          <FormInput
            inputStyle={
              Object {
                "color": "black",
              }
            }
            multiline={true}
            normalizeFontSize={true}
            onChangeText={[Function]}
            textAlign="center"
          />
          <FormLabel
            labelStyle={
              Object {
                "color": "red",
              }
            }
          >
            Product Category (/-separated)
          </FormLabel>
          <FormInput
            inputStyle={
              Object {
                "color": "black",
              }
            }
            normalizeFontSize={true}
            onChangeText={[Function]}
            textAlign="center"
          />
          <FormLabel
            labelStyle={
              Object {
                "color": "red",
              }
            }
          >
            Product Condition
          </FormLabel>
          <Picker
            mode="dialog"
            onValueChange={[Function]}
            selectedValue="New"
          >
            <PickerItem
              label="New"
              value="New"
            />
            <PickerItem
              label="Used"
              value="Used"
            />
            <PickerItem
              label="Like New"
              value="Like New"
            />
          </Picker>
          <FormLabel
            labelStyle={
              Object {
                "color": "red",
              }
            }
          >
            Who pays for shipping?
          </FormLabel>
          <Picker
            mode="dialog"
            onValueChange={[Function]}
            selectedValue="Buyer"
          >
            <PickerItem
              label="Buyer"
              value="Buyer"
            />
            <PickerItem
              label="Seller"
              value="Seller"
            />
          </Picker>
        </Card>
        <Button
          buttonStyle={
            Object {
              "backgroundColor": "#d14f4f",
            }
          }
          onPress={[Function]}
          title="Crunch the numbers!"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          

        </Text>
      </KeyboardAvoidingView>,
      "contentContainerStyle": Object {
        "backgroundColor": "#e1e8e6",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "behavior": "padding",
        "children": Array [
          <Card
            title="Add a New Listing"
          >
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Product Name
            </FormLabel>
            <FormInput
              inputStyle={
                Object {
                  "color": "black",
                }
              }
              normalizeFontSize={true}
              onChangeText={[Function]}
              textAlign="center"
            />
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Product Description
            </FormLabel>
            <FormInput
              inputStyle={
                Object {
                  "color": "black",
                }
              }
              multiline={true}
              normalizeFontSize={true}
              onChangeText={[Function]}
              textAlign="center"
            />
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Product Category (/-separated)
            </FormLabel>
            <FormInput
              inputStyle={
                Object {
                  "color": "black",
                }
              }
              normalizeFontSize={true}
              onChangeText={[Function]}
              textAlign="center"
            />
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Product Condition
            </FormLabel>
            <Picker
              mode="dialog"
              onValueChange={[Function]}
              selectedValue="New"
            >
              <PickerItem
                label="New"
                value="New"
              />
              <PickerItem
                label="Used"
                value="Used"
              />
              <PickerItem
                label="Like New"
                value="Like New"
              />
            </Picker>
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Who pays for shipping?
            </FormLabel>
            <Picker
              mode="dialog"
              onValueChange={[Function]}
              selectedValue="Buyer"
            >
              <PickerItem
                label="Buyer"
                value="Buyer"
              />
              <PickerItem
                label="Seller"
                value="Seller"
              />
            </Picker>
          </Card>,
          <Button
            buttonStyle={
              Object {
                "backgroundColor": "#d14f4f",
              }
            }
            onPress={[Function]}
            title="Crunch the numbers!"
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            

          </Text>,
        ],
        "keyboardVerticalOffset": 0,
        "style": Object {
          "backgroundColor": "#e1e8e6",
          "flex": 1,
          "justifyContent": "flex-start",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Product Name
              </FormLabel>,
              <FormInput
                inputStyle={
                  Object {
                    "color": "black",
                  }
                }
                normalizeFontSize={true}
                onChangeText={[Function]}
                textAlign="center"
              />,
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Product Description
              </FormLabel>,
              <FormInput
                inputStyle={
                  Object {
                    "color": "black",
                  }
                }
                multiline={true}
                normalizeFontSize={true}
                onChangeText={[Function]}
                textAlign="center"
              />,
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Product Category (/-separated)
              </FormLabel>,
              <FormInput
                inputStyle={
                  Object {
                    "color": "black",
                  }
                }
                normalizeFontSize={true}
                onChangeText={[Function]}
                textAlign="center"
              />,
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Product Condition
              </FormLabel>,
              <Picker
                mode="dialog"
                onValueChange={[Function]}
                selectedValue="New"
              >
                <PickerItem
                  label="New"
                  value="New"
                />
                <PickerItem
                  label="Used"
                  value="Used"
                />
                <PickerItem
                  label="Like New"
                  value="Like New"
                />
              </Picker>,
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Who pays for shipping?
              </FormLabel>,
              <Picker
                mode="dialog"
                onValueChange={[Function]}
                selectedValue="Buyer"
              >
                <PickerItem
                  label="Buyer"
                  value="Buyer"
                />
                <PickerItem
                  label="Seller"
                  value="Seller"
                />
              </Picker>,
            ],
            "title": "Add a New Listing",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Product Name",
                "labelStyle": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "Product Name",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "inputStyle": Object {
                  "color": "black",
                },
                "normalizeFontSize": true,
                "onChangeText": [Function],
                "textAlign": "center",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Product Description",
                "labelStyle": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "Product Description",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "inputStyle": Object {
                  "color": "black",
                },
                "multiline": true,
                "normalizeFontSize": true,
                "onChangeText": [Function],
                "textAlign": "center",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Product Category (/-separated)",
                "labelStyle": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "Product Category (/-separated)",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "inputStyle": Object {
                  "color": "black",
                },
                "normalizeFontSize": true,
                "onChangeText": [Function],
                "textAlign": "center",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Product Condition",
                "labelStyle": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "Product Condition",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <PickerItem
                    label="New"
                    value="New"
                  />,
                  <PickerItem
                    label="Used"
                    value="Used"
                  />,
                  <PickerItem
                    label="Like New"
                    value="Like New"
                  />,
                ],
                "mode": "dialog",
                "onValueChange": [Function],
                "selectedValue": "New",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "label": "New",
                    "value": "New",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "label": "Used",
                    "value": "Used",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "label": "Like New",
                    "value": "Like New",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Who pays for shipping?",
                "labelStyle": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "Who pays for shipping?",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <PickerItem
                    label="Buyer"
                    value="Buyer"
                  />,
                  <PickerItem
                    label="Seller"
                    value="Seller"
                  />,
                ],
                "mode": "dialog",
                "onValueChange": [Function],
                "selectedValue": "Buyer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "label": "Buyer",
                    "value": "Buyer",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "label": "Seller",
                    "value": "Seller",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonStyle": Object {
              "backgroundColor": "#d14f4f",
            },
            "onPress": [Function],
            "title": "Crunch the numbers!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "
",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "
",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <KeyboardAvoidingView
          behavior="padding"
          keyboardVerticalOffset={0}
          style={
            Object {
              "backgroundColor": "#e1e8e6",
              "flex": 1,
              "justifyContent": "flex-start",
            }
          }
        >
          <Card
            title="Add a New Listing"
          >
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Product Name
            </FormLabel>
            <FormInput
              inputStyle={
                Object {
                  "color": "black",
                }
              }
              normalizeFontSize={true}
              onChangeText={[Function]}
              textAlign="center"
            />
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Product Description
            </FormLabel>
            <FormInput
              inputStyle={
                Object {
                  "color": "black",
                }
              }
              multiline={true}
              normalizeFontSize={true}
              onChangeText={[Function]}
              textAlign="center"
            />
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Product Category (/-separated)
            </FormLabel>
            <FormInput
              inputStyle={
                Object {
                  "color": "black",
                }
              }
              normalizeFontSize={true}
              onChangeText={[Function]}
              textAlign="center"
            />
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Product Condition
            </FormLabel>
            <Picker
              mode="dialog"
              onValueChange={[Function]}
              selectedValue="New"
            >
              <PickerItem
                label="New"
                value="New"
              />
              <PickerItem
                label="Used"
                value="Used"
              />
              <PickerItem
                label="Like New"
                value="Like New"
              />
            </Picker>
            <FormLabel
              labelStyle={
                Object {
                  "color": "red",
                }
              }
            >
              Who pays for shipping?
            </FormLabel>
            <Picker
              mode="dialog"
              onValueChange={[Function]}
              selectedValue="Buyer"
            >
              <PickerItem
                label="Buyer"
                value="Buyer"
              />
              <PickerItem
                label="Seller"
                value="Seller"
              />
            </Picker>
          </Card>
          <Button
            buttonStyle={
              Object {
                "backgroundColor": "#d14f4f",
              }
            }
            onPress={[Function]}
            title="Crunch the numbers!"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            

          </Text>
        </KeyboardAvoidingView>,
        "contentContainerStyle": Object {
          "backgroundColor": "#e1e8e6",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "behavior": "padding",
          "children": Array [
            <Card
              title="Add a New Listing"
            >
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Product Name
              </FormLabel>
              <FormInput
                inputStyle={
                  Object {
                    "color": "black",
                  }
                }
                normalizeFontSize={true}
                onChangeText={[Function]}
                textAlign="center"
              />
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Product Description
              </FormLabel>
              <FormInput
                inputStyle={
                  Object {
                    "color": "black",
                  }
                }
                multiline={true}
                normalizeFontSize={true}
                onChangeText={[Function]}
                textAlign="center"
              />
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Product Category (/-separated)
              </FormLabel>
              <FormInput
                inputStyle={
                  Object {
                    "color": "black",
                  }
                }
                normalizeFontSize={true}
                onChangeText={[Function]}
                textAlign="center"
              />
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Product Condition
              </FormLabel>
              <Picker
                mode="dialog"
                onValueChange={[Function]}
                selectedValue="New"
              >
                <PickerItem
                  label="New"
                  value="New"
                />
                <PickerItem
                  label="Used"
                  value="Used"
                />
                <PickerItem
                  label="Like New"
                  value="Like New"
                />
              </Picker>
              <FormLabel
                labelStyle={
                  Object {
                    "color": "red",
                  }
                }
              >
                Who pays for shipping?
              </FormLabel>
              <Picker
                mode="dialog"
                onValueChange={[Function]}
                selectedValue="Buyer"
              >
                <PickerItem
                  label="Buyer"
                  value="Buyer"
                />
                <PickerItem
                  label="Seller"
                  value="Seller"
                />
              </Picker>
            </Card>,
            <Button
              buttonStyle={
                Object {
                  "backgroundColor": "#d14f4f",
                }
              }
              onPress={[Function]}
              title="Crunch the numbers!"
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              

            </Text>,
          ],
          "keyboardVerticalOffset": 0,
          "style": Object {
            "backgroundColor": "#e1e8e6",
            "flex": 1,
            "justifyContent": "flex-start",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <FormLabel
                  labelStyle={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  Product Name
                </FormLabel>,
                <FormInput
                  inputStyle={
                    Object {
                      "color": "black",
                    }
                  }
                  normalizeFontSize={true}
                  onChangeText={[Function]}
                  textAlign="center"
                />,
                <FormLabel
                  labelStyle={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  Product Description
                </FormLabel>,
                <FormInput
                  inputStyle={
                    Object {
                      "color": "black",
                    }
                  }
                  multiline={true}
                  normalizeFontSize={true}
                  onChangeText={[Function]}
                  textAlign="center"
                />,
                <FormLabel
                  labelStyle={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  Product Category (/-separated)
                </FormLabel>,
                <FormInput
                  inputStyle={
                    Object {
                      "color": "black",
                    }
                  }
                  normalizeFontSize={true}
                  onChangeText={[Function]}
                  textAlign="center"
                />,
                <FormLabel
                  labelStyle={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  Product Condition
                </FormLabel>,
                <Picker
                  mode="dialog"
                  onValueChange={[Function]}
                  selectedValue="New"
                >
                  <PickerItem
                    label="New"
                    value="New"
                  />
                  <PickerItem
                    label="Used"
                    value="Used"
                  />
                  <PickerItem
                    label="Like New"
                    value="Like New"
                  />
                </Picker>,
                <FormLabel
                  labelStyle={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  Who pays for shipping?
                </FormLabel>,
                <Picker
                  mode="dialog"
                  onValueChange={[Function]}
                  selectedValue="Buyer"
                >
                  <PickerItem
                    label="Buyer"
                    value="Buyer"
                  />
                  <PickerItem
                    label="Seller"
                    value="Seller"
                  />
                </Picker>,
              ],
              "title": "Add a New Listing",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Product Name",
                  "labelStyle": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "Product Name",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "inputStyle": Object {
                    "color": "black",
                  },
                  "normalizeFontSize": true,
                  "onChangeText": [Function],
                  "textAlign": "center",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Product Description",
                  "labelStyle": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "Product Description",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "inputStyle": Object {
                    "color": "black",
                  },
                  "multiline": true,
                  "normalizeFontSize": true,
                  "onChangeText": [Function],
                  "textAlign": "center",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Product Category (/-separated)",
                  "labelStyle": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "Product Category (/-separated)",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "inputStyle": Object {
                    "color": "black",
                  },
                  "normalizeFontSize": true,
                  "onChangeText": [Function],
                  "textAlign": "center",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Product Condition",
                  "labelStyle": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "Product Condition",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <PickerItem
                      label="New"
                      value="New"
                    />,
                    <PickerItem
                      label="Used"
                      value="Used"
                    />,
                    <PickerItem
                      label="Like New"
                      value="Like New"
                    />,
                  ],
                  "mode": "dialog",
                  "onValueChange": [Function],
                  "selectedValue": "New",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "label": "New",
                      "value": "New",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "label": "Used",
                      "value": "Used",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "label": "Like New",
                      "value": "Like New",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Who pays for shipping?",
                  "labelStyle": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "Who pays for shipping?",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <PickerItem
                      label="Buyer"
                      value="Buyer"
                    />,
                    <PickerItem
                      label="Seller"
                      value="Seller"
                    />,
                  ],
                  "mode": "dialog",
                  "onValueChange": [Function],
                  "selectedValue": "Buyer",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "label": "Buyer",
                      "value": "Buyer",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "label": "Seller",
                      "value": "Seller",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "buttonStyle": Object {
                "backgroundColor": "#d14f4f",
              },
              "onPress": [Function],
              "title": "Crunch the numbers!",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "
",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "
",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
